### PR TITLE
Create RFC for FS-1011: Warn when recursive function is not tail-recursive

### PR DESCRIPTION
**Pull request by neoeinstein**

### CODE REVIEW

**Comment by forki** [comment]
Did you remove the template?

**Comment by neoeinstein** [comment]
Not the core template, no. This was one of pre-created placeholders for this RFC number (there are also ones for 1012 and 1013)

**Comment by forki** [comment]
I see. Ignore me.

File: RFCs/FS-1011-Warn-on-recursive-without-tail-call.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+# Alternatives
+[alternatives]: #alternatives
+
+* Use the reserved `tailcall` keyword rather than an attribute
+* Apply the inspection to all recusive functions
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+* What warning level should the warning be issued at?
```
Comment (dsyme) [inline RFCs/FS-1011-Warn-on-recursive-without-tail-call.md]: There are a bunch of unresolved issues here, I'll add them to the list

**Comment by dsyme** [comment]
Thanks.  I remember finding a host of other issues when trying to implement this - I'll try to dig out the work and augment the RFC appropriately (one thing I like about the RFC process is it gives a proper place to record issues!)

**Pull request merged**